addons/mail/static/src/messaging/component/thread_icon/thread_icon.xml

```diff
@@ -14,7 +14,14 @@
                     </t>
                 </t>
                 <t t-elif="thread.channel_type === 'chat' and thread.directPartner">
-                    <t t-if="thread.directPartner.im_status === 'online'">
+                    <t t-if="thread.typingMembers.includes(thread.directPartner)">
```

Comment:
In practice it will be the same for DM, but I think it would be best if the condition was "at least one typing member is different than current user", to be more generic.